src/rrdcached.c

```diff
@@ -74,8 +74,9 @@ static int value_list_to_string(char *buffer, int buffer_len,
 
   memset(buffer, '\0', buffer_len);
 
-  t = CDTIME_T_TO_TIME_T(vl->time);
-  status = snprintf(buffer, buffer_len, "%lu", (unsigned long)t);
+  t = CDTIME_T_TO_DOUBLE(vl->time);
+
+  status = snprintf(buffer, buffer_len, "%f", t);
```

Comment:
The PR description and the referenced email both talk about microseconds specifically, but this format may in fact be more precise than that. Please print as `%.6f` instead.